modules/host/contractcreation.go

```diff
@@ -2,258 +2,250 @@ package host
-			err = errors.New("challenge window is not large enough")
-			return
-		}
-		// Outputs for successful proofs need to go to the correct address.
-		if contract.ValidProofAddress != h.announcement.CoinAddress {
-			err = errors.New("coins are not paying out to correct address")
-			return
-		}
-		// Output for failed proofs needs to be the 0 address.
-		emptyAddress := consensus.CoinAddress{}
-		if contract.MissedProofAddress != emptyAddress {
-			err = errors.New("burn payout needs to go to the empty address")
-			return
-		}
+	switch {
+	// TODO: check for minheight too?
+	case terms.StartHeight > maxheight:
+		return errors.New("first window is too far in the future")
 
-		// Verify that the contract fund covers the payout and burn for the whole
-		// duration.
-		requiredFund := (h.announcement.Price + h.announcement.Burn) * consensus.Currency(duration) * consensus.Currency(fileSize)
-		if contract.Payout != requiredFund {
-			err = errors.New("ContractFund does not match the terms of service.")
-			return
-		}
+	case terms.FileSize < h.MinFilesize || terms.FileSize > h.MaxFilesize:
+		return errors.New("file is of incorrect size")
 
-		// Add enough funds to the transaction to cover the penalty half of the
-		// agreement.
-		penalty := h.announcement.Burn * consensus.Currency(fileSize) * consensus.Currency(duration)
-		id, err := h.wallet.RegisterTransaction(t)
-		if err != nil {
-			err = HostCapacityErr // hide the fact that the host is having wallet issues.
-			return
-		}
-		err = h.wallet.FundTransaction(id, penalty)
-		if err != nil {
-			err = HostCapacityErr // hide the fact that the host is having wallet issues.
-			return
-		}
-		updatedTransaction, err = h.wallet.SignTransaction(id, true)
-		if err != nil {
-			err = HostCapacityErr // hide the fact that the host is having wallet issues.
-			return
-		}
+	case terms.FileSize > uint64(h.spaceRemaining):
+		return HostCapacityErr
 
-		// Update the amount of space the host has for sale.
-		h.spaceRemaining -= int64(fileSize)
-	*/
+	case duration < h.MinDuration || duration > h.MaxDuration:
+		return errors.New("duration is out of bounds")
 
-	return
+	case terms.WindowSize < h.MinWindow:
+		return errors.New("challenge window is not large enough")
+
+	case terms.ValidProofAddress != h.CoinAddress:
+		return errors.New("coins are not paying out to correct address")
+
+	case terms.MissedProofAddress != consensus.ZeroAddress:
+		return errors.New("burn payout needs to go to the zero address")
+	// TODO: should this be <= ?
+	case terms.Price != h.Price:
+		return errors.New("price does not match host settings")
+	// TODO: should this be >= ?
+	case terms.Collateral != h.Collateral:
+		return errors.New("collateral does not match host settings")
+	}
+
+	return nil
+}
+
+// verifyContract verifies that the values in the FileContract match the
+// ContractTerms agreed upon.
+// TODO: could this just return a bool (i.e. "contract does not match terms"?)
```

Comment:
larger amounts of debugging information are usually nice. Since the code is always written, it's probably better to leave it.